File: sandbox/worker/public/ohlc.py
import requests
from sseclient import SSEClient
from worker.const import BASE_URL

URL = f"{BASE_URL}/public/ohlc"

response = requests.get(URL)
print(response.text)

response = SSEClient(f"{URL}/sse")
for event in response:
    if event.data:
        print(event.data)

File: server/worker/src/api/public/ohlc.rs
use std::{convert::Infallible, time::Duration};

use axum::{
    extract::State,
    response::sse::{Event, Sse},
};
use futures::stream::{self, Stream};
use tokio_stream::StreamExt;

use crate::models::AppState;

// Return ohlc stream processed by external thread subscribed to trades
pub async fn root(
    State(state): State<AppState>,
) -> Sse<impl Stream<Item = Result<Event, Infallible>>> {
    // A `Stream` that repeats an event every second
    let stream = stream::repeat_with(|| Event::default().data("hi it is ohlc endpoint"))
        .map(Ok)
        .throttle(Duration::from_secs(1));

    Sse::new(stream).keep_alive(
        axum::response::sse::KeepAlive::new()
            .interval(Duration::from_secs(1))
            .text("keep-alive-text"),
    )
}
